### ISSUE TITLE
如何解决detach不能被赋值的问题？

### ISSUE THREAD

**Issue by MARKLIN1000** [PaddlePaddle/Paddle#43605]
### 请提出你的问题 Please ask your question

![AC423I1I46L H6 @CJ${U~A](https://user-images.githubusercontent.com/82072274/174207293-1c4c5902-f639-4c5f-a45c-e72e8c0b27ec.png)

为了操作tensor而不被autograd记录，选用了detach（没有tensor.data），但是他无法被赋值？怎么解决这个我问题？

**Comment by yt605155624** [comment]
你好，没有见过对等式左值进行 detach 的操作
已经使用了 `m = yr.detach().clone()`, 之后再 x=m, 应该就不会记录梯度了
可以参考 https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/Tensor_cn.html#detach

**Comment by hua-zi** [comment]
> 你好，没有见过对等式左值进行 detach 的操作 已经使用了 `m = yr.detach().clone()`, 之后再 x=m, 应该就不会记录梯度了 可以参考 https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/Tensor_cn.html#detach

```
import paddle
x = paddle.to_tensor([1.0, 2.0,3.0], stop_gradient=False)
yr = paddle.to_tensor([1.0, 9.0, 3.0], stop_gradient=False)
m = yr.detach().clone()
x.detach()[:] = m[:]
# x.set_value(m)
print(x)
```
请问使用 `x.detach()[:] = m[:] ` 或者 ` x.set_value(m)` 可以不被autograd记录吗
